Add cmake build configuration

p12tic opened the pull request:
cc @simogasp

simogasp commented on CMakeLists.txt:
```diff
@@ -0,0 +1,130 @@
+
+cmake_minimum_required(VERSION 3.11)
+
+project(Osi LANGUAGES C CXX)
+
+set(PROJECT_VERSION 0.107.9)
+set(PROJECT_VERSION_MAJOR 0)
+set(PROJECT_VERSION_MINOR 107)
+set(PROJECT_VERSION_PATCH 9)
```
```suggestion
project(Osi LANGUAGES C CXX VERSION 0.107.9)
```
like that CMake automatically generates the various `PROJECT_VERSION*` variables

p12tic replied:
Fixed now.

simogasp commented on CMakeLists.txt:
```diff
@@ -0,0 +1,130 @@
+find_package(CoinUtils REQUIRED)
+find_package(Mosek)
+
+file(GLOB OSI_SOURCES src/Osi/*.hpp src/Osi/*.cpp)
```
A bit of nitpicking but I'm not fan of the GLOB thing, it's always better to have something explicit like
```cmake
set(OSI_HEADERS
    "src/Osi/blabla.h"
    "...)

set(OSI_SOURCES
    "src/Osi/blabla.c"
    ...)

add_library(${PROJECT_NAME} ${OSI_SOURCES} ${OSI_HEADERS})

set_target_properties(${PROJECT_NAME}  PROPERTIES
    PUBLIC_HEADER ${OSI_HEADERS} ...)
```

p12tic replied:
Fixed now.

p12tic committed:
Add cmake configuration from vcpkg

The cmake configuration has been taken from vcpkg commit
207fbcb55cef3544f3314ef783d169419f251b3f.

p12tic committed:
Fix paths in CMake configuration

p12tic committed:
Add CMake configuration for OsiMsk library

p12tic committed:
Don't compile OsiMsk if mosek library does not exist

p12tic committed:
Use standard install paths for CMake files

p12tic committed:
Cleanup project definition for CMake

p12tic committed:
Don't use GLOB to get list of files to compile in CMake

p12tic committed:
Clean up compile options setup in CMake

p12tic commented:
@simogasp I've addressed your comments

simogasp reviewed (approved):
great!
